File: Config/GameConfig.cs
using System.Threading;
using System.Threading.Tasks;
using System.Collections.Generic;
using Firebase.Database;
using Firebase.Database.Query;

namespace MauMau.Web.Controllers
{
    public class GameConfig
    {
        public static List<string> playerNames = new List<string>() { "Alice", "Bob", "Carol", "Eva" };
        public static int numberOfCardsAtStart = 7;

        public static FirebaseClient GetFirebaseClient()
        {
            string auth = ""; // your app secret - REMOVED
            string dbUrl = "https://arduinioio.firebaseio.com/";

            FirebaseClient firebase = new FirebaseClient(dbUrl, new FirebaseOptions
            {
                AuthTokenAsyncFactory = () => Task.FromResult(auth)
            });
            return firebase;
        }


    }
}

File: ClientApp/src/config/gameConfig.ts
// if enabled, will send the game state to firebase realtime database
// and to enable Spectator mode
// leave it to false since the Firebase DB Api Secret key is removed from the server code
export const enableServerLog = false;

// enable this to use the dotnet core server for state management
// *** MauController has thread.sleep(3-5sec) to simulate slow server and to show the loading on client ***
export const useServerState = false;

//leave empty if the dotnet core server is on same host:port
export const apiHost = ''; 


export const playerNames = ['Alice', 'Bob', 'Carol', 'Eva'];

export const numberOfCardsAtStart = 7;

export const firebaseConfig = {
  // apiKey: "",  // not needed for client app
  authDomain: "arduinioio.firebaseio.com/",
  databaseURL: "https://arduinioio.firebaseio.com/",
}

// TODO: put this setting in gameState object
let hasCpuPlayers = false;
export const setHasCpuPlayers = (newVal: boolean) => hasCpuPlayers = newVal;
export const getHasCpuPlayers = () => hasCpuPlayers;
